include/envoy/stream_info/stream_info.h

```diff
@@ -64,6 +65,50 @@ enum ResponseFlag {
   LastFlag = StreamIdleTimeout
 };
 
+struct ResponseCodeDetailValues {
```

Comment:
Those look great, thanks, but you're effectively rewriting even the existing `Proxy-Status` types, so what's your plan in regards to that header? Using corresponding standard types and then using those values in the `details` field or completely ignoring standard types and only using `http_{request,response}_error` with those values in `details`? In either case, I think you might need a `std::pair` (or perhaps even `std::tuple`) here, instead of only strings.

Reply:
I think when we implement proxy-status these could be extended.  I could do a one off with today's proxy-status codes but I really don't want to sign up for keeping it up to date and I'd prefer to not add code which is going to bitrot as the IETF makes forward progress.  I don't think creating this without the tuple creates more work for when we implement proxy-status, and it might reduce it (by filling in the details field)